scripts/task_process/cache_status.py

```diff
@@ -193,23 +193,40 @@ def parseErrorReport(data, nodes):
             # (even if there's only a single value) it has to be indexed to zero.
             statedict['Error'] = data[jobid].values()[0] #data[jobid] contains all retries. take the last one
 
-def parseNodeStateV2(fp, nodes):
+def parseNodeStateV2(fp, nodes, level):
     """
     HTCondor 8.1.6 updated the node state file to be classad-based.
     This is a more flexible format that allows future extensions but, unfortunately,
     also requires a separate parser.
     """
-    taskStatus = nodes.setdefault("DagStatus", {})
+    dagStatus = nodes.setdefault("DagStatus", {})
+    dagStatus.setdefault("SubDagStatus", {})
+    subDagStatus = dagStatus.setdefault("SubDags", {})
     for ad in classad.parseAds(fp):
         if ad['Type'] == "DagStatus":
-            taskStatus['Timestamp'] = ad.get('Timestamp', -1)
-            taskStatus['NodesTotal'] = ad.get('NodesTotal', -1)
-            taskStatus['DagStatus'] = ad.get('DagStatus', -1)
+            if level:
+                statusDict = subDagStatus.setdefault(int(level), {})
+                statusDict['Timestamp'] = ad.get('Timestamp', -1)
+                statusDict['NodesTotal'] = ad.get('NodesTotal', -1)
+                statusDict['DagStatus'] = ad.get('DagStatus', -1)
+            else:
+                dagStatus['Timestamp'] = ad.get('Timestamp', -1)
```

Comment:
these 3 lines seems to me the same before and after the else. Could they be put out of the if ?

Reply:
Not sure that I understand your comment? One sets keys in `dagStatus` and one in `statusDict`?

Reply:
sorryt. I was confused. You are right. please forgive and forget.